source/adapters/cuda/command_buffer.hpp

```diff
@@ -175,6 +175,85 @@ static inline const char *getUrResultString(ur_result_t Result) {
       fprintf(stderr, "UR <--- %s(%s)\n", #Call, getUrResultString(Result));   \
   }
 
+// Handle to a kernel command.
+//
+// Struct that stores all the information related to a kernel command in a
+// command-buffer, such that the command can be recreated. When handles can
+// be returned from other command types this struct will need refactored.
+struct ur_exp_command_buffer_command_handle_t_ {
+  ur_exp_command_buffer_command_handle_t_(
+      ur_exp_command_buffer_handle_t CommandBuffer, ur_kernel_handle_t Kernel,
+      std::shared_ptr<CUgraphNode> Node, CUDA_KERNEL_NODE_PARAMS Params,
+      uint32_t WorkDim, const size_t *pGlobalWorkOffset,
+      const size_t *pGlobalWorkSize, const size_t *pLocalWorkSize)
+      : CommandBuffer(CommandBuffer), Kernel(Kernel), Node(Node),
+        Params(Params), WorkDim(WorkDim), RefCount(1) {
+    UR_TRACE(urCommandBufferRetainExp(CommandBuffer));
+
+    const size_t CopySize = sizeof(size_t) * WorkDim;
+    std::memcpy(GlobalWorkOffset, pGlobalWorkOffset, CopySize);
+    std::memcpy(GlobalWorkSize, pGlobalWorkSize, CopySize);
+    std::memcpy(LocalWorkSize, pLocalWorkSize, CopySize);
+
+    if (WorkDim < 3) {
+      const size_t ZeroSize = sizeof(size_t) * (3 - WorkDim);
+      std::memset(GlobalWorkOffset + WorkDim, 0, ZeroSize);
+      std::memset(GlobalWorkSize + WorkDim, 0, ZeroSize);
+      std::memset(LocalWorkSize + WorkDim, 0, ZeroSize);
+    }
+  }
+
+  ~ur_exp_command_buffer_command_handle_t_() {
+    UR_TRACE(urCommandBufferReleaseExp(CommandBuffer));
+  }
+
+  void SetGlobalOffset(const size_t *pGlobalWorkOffset) {
+    const size_t CopySize = sizeof(size_t) * WorkDim;
+    std::memcpy(GlobalWorkOffset, pGlobalWorkOffset, CopySize);
+    if (WorkDim < 3) {
+      const size_t ZeroSize = sizeof(size_t) * (3 - WorkDim);
+      std::memset(GlobalWorkOffset + WorkDim, 0, ZeroSize);
+    }
+  }
+
+  void SetGlobalSize(const size_t *pGlobalWorkSize) {
+    const size_t CopySize = sizeof(size_t) * WorkDim;
+    std::memcpy(GlobalWorkSize, pGlobalWorkSize, CopySize);
+    if (WorkDim < 3) {
+      const size_t ZeroSize = sizeof(size_t) * (3 - WorkDim);
+      std::memset(GlobalWorkSize + WorkDim, 0, ZeroSize);
+    }
+  }
+
+  void SetLocalSize(const size_t *pLocalWorkSize) {
+    const size_t CopySize = sizeof(size_t) * WorkDim;
+    std::memcpy(LocalWorkSize, pLocalWorkSize, CopySize);
+    if (WorkDim < 3) {
+      const size_t ZeroSize = sizeof(size_t) * (3 - WorkDim);
+      std::memset(LocalWorkSize + WorkDim, 0, ZeroSize);
+    }
+  }
+
+  // Used when retaining an object.
+  uint32_t incrementReferenceCount() noexcept { return ++RefCount; }
+  // Used when releasing an object.
+  uint32_t decrementReferenceCount() noexcept { return --RefCount; }
+  uint32_t getReferenceCount() const noexcept { return RefCount; }
```

Comment:
Just a nit here but prefer to use consistent naming convention, i.e. all of the class' functions and variable members starting with upper case following PascalCase style.

Reply:
this is correct for llvm style, variables are Pascal and functions are camel

Reply:
Some of the above functions are Pascal

Reply:
Yeah that's fine with me. The nit comment was rather me pointing out the inconsistency between the function members of the struct where the setters are _Pascal_ style while those getters are _camel_ style. So long as all the member functions follow the correct naming convention, in this case being that from the llvm style, that'll be grand. Thanks.

Reply:
yep didn't clock the other functions, always get more context than what you see in the email :facepalm:

Reply:
Good spot, I've updated the other functions defined in this file to llvm style